### ISSUE TITLE
undesirable_function_linter could handle setter functions better

### ISSUE THREAD

**Issue by MichaelChirico** [r-lib/lintr#1912]
This might be considered a false positive:

```r
linter <- lintr::undesirable_function_linter(c("time" = NA))
lintr::lint("
time(x) <- time(y)
", linter)
```

Throws a lint for both LHS & RHS, but technically on the LHS we're calling `time<-`, not `time()` _per se_.

This mattered for me because `time()` is `stats::time()`, but `time<-` is ``zoo::`time<-` ``, not ``stats::`time<-` ``.
